xdsl/parser/attribute_parser.py

```diff
@@ -1250,6 +1250,7 @@ def _parse_optional_integer_or_float_type(self) -> Attribute | None:
         Parse as integer or float type, if present.
           integer-or-float-type ::= index-type | integer-type | float-type
           index-type            ::= `index`
+          unit-attribute        ::= `unit`
```

Comment:
I first tried to add this as a normal attribute not type, but then the test failed, saying "type expected". I have a supsicion it's actually a type in MLIR also, but my quick search for the definition didn't yield much so I thought I'd open a PR anyway.

Reply:
It is not a type, it's an attribute though: https://mlir.llvm.org/docs/Dialects/Builtin/#unitattr

Reply:
Should I change the test instead? When I put the code for parsing into `parse_attribute` it fails, as the block argument expects a type.

Reply:
Which test? And yes, we should change it I think